### ISSUE TITLE
Need unit tests

### ISSUE THREAD

**Issue by philparadis** [micro222/chatbot#8]
Debugging is kinda painful now. Unit tests would be welcome. We don't have to go crazy about it especially since we don't have any framework to run, but I would like a basic set of unit tests where I can just run ./unit_tests_chatbot and get a list of OK's and ERROR's that give me a rough idea of what broke when I changed something.

**Comment by micro222** [comment]
I can't say I know what unit tests are, but they do sound useful :-)  How can I help?

**Comment by philparadis** [comment]
Whenever you write a function, think of all possible input to the functions and what should be the correct matching output.

Then write a series of tests that write like

test_equal(my_func(1, 2), 3)
test_equal(my_func(-1, -1), -2)
test_error(my_func("abc", 1))

They should all print "Test OK" or "Test FAILED" to the screen and report details.

Also, if the function can receive used input, you might want more tests for the higher possibilities of having human input something invalid or even worse, something malicious.

Basically, we'll need to write a unit test framework before getting started on this. This should be a side project as a branch, to be merged with the main trunk later.
